### ISSUE TITLE
[Feature/suggestion] handling of remote items

### ISSUE THREAD

**Issue by yyoncho** [noctuid/targets.el#26]
I believe that the "remote word" text object would have a better user experience it is implemented using `avy-goto-word-1` (or even making the selection of `avy-goto-word-N` method configurable), what do you think?

**Comment by noctuid** [comment]
Targets (and things.el) is/will be a generic interface built on top of thingatpt. Seeking with avy will be done in the same way for all things. I'll look at these avy commands to see if remote selection might be improved in general. I could potentially allow direct customization of a thing's avy selection function, but my guess is that this is unnecessary.

**Comment by noctuid** [comment]
Yeah, I think it would be a great idea to provide some way to (just like for motions) pre-filter things to prevent an explosion of overlays. Thanks for the suggestion. 

I'll add a way of running a predicate on each thing to see if it should be included that will allow prompting the user for the first n characters, for example. Separate `...word-N` things/text objects could then be created with the behavior of `avy-goto-word-N` for selection.

**Comment by noctuid** [comment]
This is now implemented in things.el and will eventually be possible in targets (once it is rewritten to use things.el).

**Comment by yyoncho** [comment]
Great! I was thinking about one more feature(or maybe optional behavior): the main advantage of remote objects is you can act on them from a remote location without changing your position. AFAIK that is true for all of the operators except change. What do you think about making `change` operator go back to the starting position when you switch to normal state? Of course, this could be implemented on top of targets, but I wonder whether it could be part of the `targets.el` core?

**Comment by noctuid** [comment]
Yes, that's another good idea. I'll add it as an option but not as the default when I get the chance. Feel free to make more issues with suggestions.
